newrelic-agent/src/main/java/com/newrelic/agent/config/ObscuringConfig.java

```diff
@@ -40,7 +38,7 @@ public Object apply(Object input) {
                 if (obscuringKey != null) {
                     return ((ObscuredYamlPropertyWrapper) input).getValue(obscuringKey);
                 }
-                AgentBridge.getAgent().getLogger().log(Level.WARNING, "Unable to deobfuscate value. Missing a obscuring key");
+                throw new IllegalArgumentException("Missing the obscuring key");
```

Comment:
can we make this more descriptive, I liked the warning message above. Also could we make it a RuntimeException?

Reply:
sounds good, changed!